CoreStubs/ArchiveControl/module.php

```diff
@@ -23,6 +29,26 @@ private function SetVariableData($VariableID, $Data)
         $this->Archive[$VariableID] = $Data;
     }
 
+    public function StubsAddAggregatedValues(int $VariableID, int $AggregationSpan, array $AggregationData)
+    {
+        if (!$this->GetLoggingStatus($VariableID)) {
+            throw new Exception('Adding aggregated data requires active logging');
+        }
+        usort($AggregationData, function ($a, $b)
+        {
+            return $a['TimeStamp'] <=> $b['TimeStamp'];
+        });
+
+        $AggregatedArchiveData = $this->GetVariableData($VariableID, $AggregationSpan)['AggregatedValues'][$AggregationSpan];
+
+        if ((count($AggregatedArchiveData) > 0) && (count($AggregationData) > 0) &&
+                ($AggregationData[0]['TimeStamp'] < $AggregatedArchiveData[count($AggregatedArchiveData) - 1]['TimeStamp'])) {
+            throw new Exception('It is not yet possible to add aggregated values before the newest');
+        }
+
+        $this->Archive[$VariableID]['AggregatedValues'][$AggregationSpan] = array_merge($AggregatedArchiveData, $AggregationData);
```

Comment:
Wir haben doch eine "$this->SetVariableData($VariableID, $ArchivedData);" Funktion. Ich finde die sollten wir hier auch nutzen, oder?